server/consumer.go

```diff
@@ -961,6 +961,7 @@ func (o *Consumer) loopAndDeliverMsgs() {
 			c.pa.size = len(pm.msg) + len(pm.hdr)
 			c.pa.szb = []byte(strconv.Itoa(c.pa.size))
 			c.pa.reply = []byte(pm.reply)
+			c.pa.topic = []byte(pm.topic)
```

Comment:
What exactly is the role of the topic here?

Reply:
When parsing an MQTT publish message, there is a topic, say "foo/bar". It is converted to foo.bar to do sublist match, etc.. but when it is time to deliver to MQTT subscriptions, if we have only `subject` which is foo.bar, we would need to convert foo.bar back to foo/bar. Instead, we save reference to topic foo/bar in c.pa.topic. However, for QoS1 when JS is the publisher (internal client) we pass the jsPubMsg.topic to c.pa.topic for processing in mqttDeliverMsgCb().
How we get it into jsPubMsg? This is in processInboundJetStreamMsg() where we pass pc.pa.topic if not empty.